lib/os/reboot.c

```diff
@@ -24,3 +25,13 @@ FUNC_NORETURN void sys_reboot(int type)
 		k_cpu_idle();
 	}
 }
+
+int __weak sys_arch_get_reboot_type(void)
+{
+	return SYS_REBOOT_WARM;
```

Comment:
This seems like an incorrect default implementation. I'm not exactly sure what the right thing to do is, though. But perhaps you could add a new SYS_REBOOT_UNKNOWN and define it to -1 or something like that?